packages/orama/src/components/sorter.ts

```diff
@@ -103,56 +101,99 @@ async function insert(
   if (!sorter.enabled) {
     return
   }
+
+  sorter.language = language
+  sorter.isSorted = false
+
   const s = sorter.sorts[prop]
 
-  let predicate: (value: [string, SortValue]) => boolean
-  switch (schemaType) {
+  s.docs.set(id, s.orderedDocs.length);
+  s.orderedDocs.push([id, value]);
+}
+
+function ensureIsSorted(sorter: Sorter): void {
+  if (sorter.isSorted) {
+    return
+  }
+
+  if (!sorter.enabled) {
+    return
+  }
+
+  for (const prop of Object.keys(sorter.sorts)) {
```

Comment:
Move `Object.keys` outside the loop condition to avoid accessing the object keys on every iteration